Extensions/probing.hh

```diff
@@ -158,132 +158,81 @@ inline double Potential(double data, std::vector< std::vector<double> > params,
+                                                 0.374470347084, 1.27932240423,
+                                                 0.0, 0.374470347084,
+                                                 1.27932240423, 0.0}};
+
+  static constexpr double DMS_params[2][8] = {{1.36184674022, 0.0,
+                                               0.0876565404957, 1.36184674022,
+                                               0.0, 0.0876565404957,
+                                               1.36184674022, 0.0},
+                                              {1.33486621438, 0.0,
+                                               0.37015874678, 1.33486621438,
+                                               0.0, 0.37015874678,
+                                               1.33486621438, 0.0}};
+
+  static constexpr double CMCT_params[2][8] = {{0.668918986169, 0.0,
+                                                0.268161495459, 0.668918986169,
+                                                0.0, 0.268161495459,
+                                                0.668918986169, 0.0},
+                                               {0.641092593747, 0.0,
+                                                0.8373230903, 0.641092593747,
+                                                0.0, 0.8373230903,
+                                                0.641092593747, 0.0}};
+
+  if (data <= -500) {
+    return 0;
   }
 
-  if (data <= -500)
-    return 0;
   if (modifier == "SHAPE_AC" || modifier == "SHAPE_GU") {
     // This is only applied if SHAPE_AC or SHAPE_GU is specified
     // For now, I'm using the "default" calculations for SHAPE
     // pseudoenergies when the modifier is "SHAPE".
     params = SHAPE_params;
+  } else if (modifier == "DMS") {
+    params = DMS_params;
+  } else if (modifier == "CMCT") {
+    params = CMCT_params;
+  } else if (modifier == "diffSHAPE") {
+    if (data > 0) {
+      return data * slope;
+    } else {
+      return 0;
+    }
   } else {
-    if (modifier == "DMS") {
-      params = DMS_params;
+    if (data > 0) {
+      return log(data + 1.0) * slope + intercept;
     } else {
-      if (modifier == "CMCT") {
-        params = CMCT_params;
-      } else {
-        if (modifier == "diffSHAPE") {
-          if (data > 0) {
-            return data * slope;
-          } else {
-            return 0;
-          }
-        } else {
-          if (data > 0) {
-            return log(data + 1.0) * slope + intercept;
-          } else {
-            return intercept;
-          }
-        }
-      }
+      return intercept;
     }
   }
-  if (data < 0 || (slope == 0 && intercept == 0))
+
+  if (data < 0 || (slope == 0 && intercept == 0)) {
     return 0;
+  }
   // double val2 = log(data+1.0)*slope+intercept;
   double kT = 5.904976983149999;
   double val = Potential(data, params, kT);
 
-  if ((slope == 0 && intercept == 0)) {
-    return 0;
-  } else {
-    return val;
-  }
```

Comment:
it's not easy to understand the logic in the diff view, but is it correct that this condition is gone altogether?

Reply:
I believe so too and I'm not sure if this might be the reason for the different results regarding the tests?

Reply:
I removes this if statement, because the condition is already checked once above this statement around line 225:
```
if (data < 0 || (slope == 0 && intercept == 0)) {
    return 0;
  }
```
If this condition were to be true, the function would already return 0 here, so the if statement you are referring to would never be reached, which is why I removed it.

Reply:
I don't know enough about c++ but didn't you forget an `else { return val; }` ?
Or is the `return val;`in line 232 only executed if all the other conditions are false?

Reply:
But your other point is completely true of course